src/test/kotlin/am/ik/yavi/builder/ValidatorBuilderKtDslTest.kt

```diff
@@ -0,0 +1,1032 @@
+        Assertions.assertThat(violations.isValid).isFalse()
+        Assertions.assertThat(violations.size).isEqualTo(1)
+        violation = violations[0]
+        Assertions.assertThat(violation.message()).isEqualTo("""The size of "x" must be less than 5. The given size is 6""")
+        Assertions.assertThat(violation.messageKey()).isEqualTo("container.lessThan")
+    }
+
+    @Test
+    fun constraintOnCharSequenceIfPresent() {
+        val validator = validator<DemoString> {
+            DemoString::x {
+                lessThan(5)
+            }
+        }
+
+        var demo = DemoString("foo")
+        var violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isTrue()
+
+        demo = DemoString(null)
+        violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isTrue()
+
+        demo = DemoString("")
+        violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isTrue()
+
+        demo = DemoString("foofoo")
+        violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isFalse()
+        Assertions.assertThat(violations.size).isEqualTo(1)
+        var violation = violations[0]
+        Assertions.assertThat(violation.message()).isEqualTo("""The size of "x" must be less than 5. The given size is 6""")
+        Assertions.assertThat(violation.messageKey()).isEqualTo("container.lessThan")
+    }
+
+    @Test
+    fun constraintOnCharSequenceNotNullable() {
+        val validator = validator<DemoStringNotNullable> {
+            DemoStringNotNullable::x {
+                notEmpty()
+                lessThan(5)
+            }
+        }
+
+        var demo = DemoStringNotNullable("foo")
+        var violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isTrue()
+
+        demo = DemoStringNotNullable("")
+        violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isFalse()
+        Assertions.assertThat(violations.size).isEqualTo(1)
+        var violation = violations[0]
+        Assertions.assertThat(violation.message()).isEqualTo(""""x" must not be empty""")
+        Assertions.assertThat(violation.messageKey()).isEqualTo("container.notEmpty")
+
+        demo = DemoStringNotNullable("foofoo")
+        violations = validator.validate(demo)
+        Assertions.assertThat(violations.isValid).isFalse()
+        Assertions.assertThat(violations.size).isEqualTo(1)
+        violation = violations[0]
+        Assertions.assertThat(violation.message()).isEqualTo("""The size of "x" must be less than 5. The given size is 6""")
+        Assertions.assertThat(violation.messageKey()).isEqualTo("container.lessThan")
+    }
+
+    @Test
+    fun constraintOnCharSequenceRequired() {
+        val validator = validator<DemoString> {
+            DemoString::x required {
+                lessThan(5)
+            }
+        }
```

Comment:
how about this _required_ infix - since this is something 'additional' to the non-Kotlin-DSL can this also be added? (for all supported constraint types)

Implementation is:
```kotlin
    /** Q: this will be required for each type */
    infix fun KProperty1<T, String?>.required(block: CharSequenceConstraint<T, String?>.() -> Unit) =
        validatorBuilder.constraint(this, this.name) { it.notNull().apply(block) }
```

If I my understanding is right, since YAVI is Java first - by default all validations are _optional_ and will only be applied if the field is notNull.

Reply:
@ngsw-taro any thoughts?

Reply:
> by default all validations are optional and will only be applied if the field is notNull.

Except for `notNull`, `notBlank` and `notEmpty`, `null` is regarded as a valid input 
https://yavi.ik.am/#built-in-constraints

Reply:
it's quite experimental and different from YAVI itself, maybe better to just drop the idea?

It's obviously straightforward and clean to just do the normal way...

```kotlin
val validator = validator<DemoString> {
    DemoString::x {
        notNull()
        lessThan(5)
    }
}
```

Reply:
From a non-Kotlin user perspective, I prefer the normal way ;)

Reply:
👍  I have removed the infix fun + test